quic/transport/tlsbackend.nim

```diff
@@ -1,24 +1,40 @@
 import results
-import ngtcp2/native/picotls
 import ngtcp2
+import ./ngtcp2/native
 import ../errors
 
+export CertificateVerifier
+export certificateVerifierCB
+export CustomCertificateVerifier
+export InsecureCertificateVerifier
+export init
+export destroy
+
 type TLSBackend* = ref object
   picoTLS*: PicoTLSContext
 
-proc init*(
-    t: typedesc[TLSBackend], isServer: bool, certificate: seq[byte], key: seq[byte]
+proc newServerTLSBackend*(
+    certificate: seq[byte],
+    key: seq[byte],
+    certificateVerifier: Opt[CertificateVerifier],
 ): TLSBackend {.raises: [QuicError].} =
-  let picotlsCtx = PicoTLSContext.init(certificate, key)
-  if isServer:
-    let ret = ngtcp2_crypto_picotls_configure_server_context(picotlsCtx.context)
-    if ret != 0:
-      raise newException(QuicError, "could not configure server context: " & $ret)
-  else:
-    let ret = ngtcp2_crypto_picotls_configure_client_context(picotlsCtx.context)
-    if ret != 0:
-      raise newException(QuicError, "could not configure client context: " & $ret)
+  let picotlsCtx = PicoTLSContext.init(
+    certificate, key, certificateVerifier, certificateVerifier.isSome
+  )
+  let ret = ngtcp2_crypto_picotls_configure_server_context(picotlsCtx.context)
+  if ret != 0:
+    raise newException(QuicError, "could not configure server context: " & $ret)
+  return TLSBackend(picoTLS: picotlsCtx)
 
+proc newClientTLSBackend*(
+    certificate: seq[byte],
+    key: seq[byte],
+    certificateVerifier: Opt[CertificateVerifier],
+): TLSBackend {.raises: [QuicError].} =
```

Comment:
not sure if this should raise `QuicError`

Reply:
Done in last commit!